scss/mixins/_forms.scss

```
@@ -129,7 +129,7 @@
 
   .input-group .form-control,
   .input-group .form-select {
-    @include form-validation-state-selector($state) {
+    @if ($state == "invalid") {
```

```suggestion
    @include form-validation-state-selector($state) {
      @if $state == "valid" {
        z-index: 2;
      } @else if $state == "invalid" {
        z-index: 3;
      }
    }
```

Should do the trick, I think.